src/features/berth/berthFormPage/boatDetails/BoatDetails.tsx

```diff
@@ -0,0 +1,97 @@
+import { useState, useEffect } from 'react';
+import { useForm } from 'react-final-form';
+
+import { WithBoatType } from '../../../../common/selects/Selects';
+import { BerthFormValues, Harbors, Boat } from '../../types';
+import CreateBerthBoat from './createBerthBoat/CreateBerthBoat';
+import BoatDetailsSelectBoat from './BoatDetailsSelectBoat';
+import BoatDetailsVerifyBoat from './BoatDetailsVerifyBoat';
+
+enum BoatDetailsPhase {
+  // Trying to find what phase is most fitting for the user.
+  INDETERMINATE = 1,
+  // User has many boats of which they have to select one.
+  SELECT_BOAT = 2,
+  // User checks the information of the boat. User is sent
+  // into this phase automatically if they only have a single
+  // boat.
+  VERIFY_BOAT = 3,
+  // User can add a new boat in case they do not have
+  // a boat or the boat they are making an application for
+  // is not added yet.
+  CREATE_BOAT = 4,
+}
+
+function getInitialPhase(boats?: any[]) {
```

Comment:
```suggestion
function getInitialPhase(boats?: Boat[]) {
```
If the type doesn't matter, I would rather use `unknown`